frontend/apps/reader/modules/readeruserhyph.lua

```diff
@@ -0,0 +1,228 @@
+        end,
+        checked_func = function()
+            return self:isAvailable()
+        end,
+        enabled_func = function()
+            return self:_enabled()
+        end,
+        separator = true,
+    }
+end
+
+-- Helper functions for dictionary entries-------------------------------------------
+
+-- checks if suggestion is well formated
+function ReaderUserHyph:checkHyphenation(suggestion, word)
+    if suggestion:find("%-%-") then
+        return false -- two or more consecutive '-'
+    end
+
+    suggestion = suggestion:gsub("-","")
+    if self.ui.document:getLowercasedWord(suggestion) == self.ui.document:getLowercasedWord(word) then
+        return true -- characters match (case insensitive)
+    end
+    return false
+end
+
+function ReaderUserHyph:updateDictionary(word, hyphenation)
+    local dict_file = self:getDictionaryPath()
+    local new_dict_file = dict_file .. ".new"
+
+    local new_dict = io.open(new_dict_file, "w")
+    if not new_dict then
+        logger.err("UserHyph: could not open " .. new_dict_file)
+        return
+    end
+
+    local word_lower = self.ui.document:getLowercasedWord(word)
+    local line
+
+    local dict = io.open(dict_file, "r")
+    if dict then
+        line = dict:read()
+        --search entry
+        while line and self.ui.document:getLowercasedWord(line:sub(1, line:find(";") - 1)) < word_lower do
+            new_dict:write(line .. "\n")
+            line = dict:read()
+        end
+
+        -- last word = nil if EOF, else last_word=word if found in file, else last_word is word after the new entry
+        if line then
+            local last_word = self.ui.document:getLowercasedWord(line:sub(1, line:find(";") - 1))
+            if last_word == self.ui.document:getLowercasedWord(word) then
+                line = nil -- word found
+            end
+        else
+            line = nil -- EOF
+        end
+    end
+
+    -- write new entry
+    if hyphenation and hyphenation ~= "" then
+        new_dict:write(string.format("%s;%s\n", word, hyphenation))
+    end
+
+    -- write old entry if there was one
+    if line then
+        new_dict:write(line .. "\n")
+    end
+
+    if dict then
+        repeat
+            line = dict:read()
+            if line then
+                new_dict:write(line .. "\n")
+            end
+        until (not line)
+        dict:close()
+        os.remove(dict_file)
+    end
+
+    new_dict:close()
+    os.rename(new_dict_file, dict_file)
+
+    self:loadUserDictionary(true) -- dictionary has changed, force a reload here
+end
+
+function ReaderUserHyph:modifyUserEntry(word)
+    if word:find("[ ,;-%.]") then return end -- no button if more than one word
+
+    if not self.ui.document then return end
+
+    local suggested_hyphenation = self.ui.document:getHyphenationForWord(word)
+
+    local input_dialog
+    input_dialog = InputDialog:new{
+        title = T(_("Hyphenation entry for: \"%1\""), word),
```

Comment:
This looks a bit long:
![image](https://user-images.githubusercontent.com/24273478/120114643-65a1a880-c180-11eb-9d2a-cfc72b0022d9.png)

Would just `Hyphenate: %1` work?

Reply:
Yes this will work.
I have never heard this word, must be german-German :)

Reply:
Store? Why not just save?